src/components/cylc/Drawer.vue

```diff
@@ -83,72 +85,55 @@ export default {
     'c-header': Header
   },
 
-  data: function () {
-    return {
-      drawerWidth: initialWidth
-    }
-  },
+  setup () {
+    const { mobile } = useDisplay()
 
-  mounted () {
-    this.setEvents()
-  },
+    const drawerWidth = ref(initialWidth)
+
+    const { drawer } = useDrawer()
+    // Show drawer initially if viewport is large enough:
+    drawer.value = !mobile.value
 
-  computed: {
-    ...mapState('user', ['user']),
-    drawer: {
-      get () {
-        return this.$store.state.app.drawer
-      },
-      set (val) {
-        this.setDrawer(val)
-      }
+    function resize (e) {
+      // If less than min width, will collapse (to 4px because that's the resize-bar width)
+      drawerWidth.value = e.clientX > minWidth ? e.clientX : 4
     }
-  },
 
-  methods: {
-    ...mapMutations('app', ['setDrawer']),
-    getDrawerElement () {
-      // Cannot use $refs.drawerRef.$el due to https://github.com/vuetifyjs/vuetify/issues/16766
-      return document.getElementById('c-sidebar')
-    },
-    resize (e) {
-      // If less than min width, will collapse (to 4px because that's the
-      // resize-bar width)
-      this.drawerWidth = e.clientX > minWidth ? e.clientX : 4
-    },
-    setEvents () {
-      const el = this.getDrawerElement()
-      const drawerBorder = this.$refs.resizeBar
-      drawerBorder.addEventListener(
+    /** @type {import('vue').Ref<HTMLElement>} template ref */
+    const resizeBar = ref(null)
+
+    when(resizeBar, () => {
```

Comment:
Using `when` waits until the template ref is available. I think we could use `onMounted` instead, but maybe not now that some Vuetify components are using [`<Suspense>`](https://vuejs.org/guide/built-ins/suspense.html#suspense) - see https://github.com/vuetifyjs/vuetify/issues/19736#issuecomment-2089912293

Reply:
Appears to work ok. The target region for resizing is very narrow and can be obscured by scroll bars, but this isn't new, opened a separate issue.